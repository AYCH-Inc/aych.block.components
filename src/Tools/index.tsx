// import * as interpolation from "./interpolation"
export { default as interpolation } from './interpolation';
export * from './testHelper';
export * from './theme';
export { default as ThemeProvider } from './ThemeProvider';
