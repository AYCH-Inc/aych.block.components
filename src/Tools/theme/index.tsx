/* eslint-disable sort-keys */
import { defaultConfigTheme, ConfigTheme } from './config';
import { defaultButtonTheme, ButtonTheme } from './buttons';
import { defaultNavbarTheme, NavbarTheme } from './navbar';
import { defaultTypographyTheme, TypographyTheme } from './typography';

export { ConfigTheme, ButtonTheme, TypographyTheme };

export interface Theme {
  config: ConfigTheme;
  button: ButtonTheme;
  navbar: NavbarTheme;
  typography: TypographyTheme;
}

export type ThemeComponentType = keyof Theme;

export type ThemePropertyType = keyof ButtonTheme | keyof TypographyTheme | keyof NavbarTheme;

export const defaultTheme: Theme = {
  config: defaultConfigTheme,
  button: defaultButtonTheme,
  navbar: defaultNavbarTheme,
  typography: defaultTypographyTheme,
};

/* eslint-enable sort-keys */
