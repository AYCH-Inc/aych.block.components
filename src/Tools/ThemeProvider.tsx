import * as React from 'react';
import { mergeRight } from 'ramda';
import { ThemeProvider as StyledComponentThemeProvider } from 'styled-components';
import { defaultTheme } from './theme';

interface Props {
  children: JSX.Element;
  theme?: object;
}

const ThemeProvider = (props: Props): JSX.Element => (
  <StyledComponentThemeProvider theme={mergeRight(props.theme, defaultTheme)}>
    {props.children}
  </StyledComponentThemeProvider>
);

ThemeProvider.defaultProps = {
  theme: defaultTheme,
};

export default ThemeProvider;
