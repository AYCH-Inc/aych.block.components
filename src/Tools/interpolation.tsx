import { pathOr } from 'ramda';
import { FontSizeType, PaletteType, ScreenSizeType, FontWeightType } from './theme/config';
import { Theme, ThemeComponentType, ThemePropertyType } from './theme';

const color: (colorKey: PaletteType) => (props: { theme: Theme }) => string = colorKey => props =>
  props.theme.config.defaultPalette[colorKey];

const theme = (component: ThemeComponentType, property: ThemePropertyType) => (props: { theme: Theme }) =>
  color(pathOr('red200', ['theme', component, property], props))(props);

const fontSize = (property: FontSizeType) => (props: { theme: Theme }): string =>
  pathOr('1rem', ['theme', 'config', 'defaultFontSizes', property], props);

const fontWeight = (property: FontWeightType) => (props: { theme: Theme }): number =>
  pathOr(400, ['theme', 'config', 'defaultFontWeights', property], props);

const screenSize = (property: ScreenSizeType) => (props: { theme: Theme }): string =>
  pathOr('48rem', ['theme', 'config', 'defaultScreenSizes', property], props);

export default {
  color,
  fontSize,
  fontWeight,
  screenSize,
  theme,
};
