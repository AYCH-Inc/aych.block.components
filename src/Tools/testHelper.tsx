import React from 'react';
import { mount, ReactWrapper } from 'enzyme';
import ThemeProvider from './ThemeProvider';

interface Props {
  children: React.ReactElement;
}

const getThemeProviderWrappingComponent = ({ children }: Props): JSX.Element => (
  <ThemeProvider>{children}</ThemeProvider>
);

export const mountWithTheme = (component: React.ReactElement): ReactWrapper => {
  const wrapper = mount(component, {
    wrappingComponent: getThemeProviderWrappingComponent,
  });

  return wrapper;
};
