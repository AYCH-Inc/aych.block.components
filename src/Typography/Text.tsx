import * as React from 'react';
import styled from 'styled-components';
import { interpolation } from '../Tools';
import { FontSizeType, FontWeightType } from '../Tools/theme/config';

interface Props {
  children: React.ReactNode;
  size: FontSizeType;
  weight: FontWeightType;
}

const TextStyle = styled.span<Props>`
  font-family: Inter, sans-serif;
  font-size: ${(props: Props) => interpolation.fontSize(props.size)};
  font-weight: ${(props: Props) => interpolation.fontWeight(props.weight)};
  color: ${interpolation.theme('typography', 'textForegroundColor')};
`;

const Text = ({ children, ...rest }: Props): JSX.Element => <TextStyle {...rest}>{children}</TextStyle>;

Text.defaultProps = {
  size: 'size5',
  weight: 'weight4',
};

export default Text;
