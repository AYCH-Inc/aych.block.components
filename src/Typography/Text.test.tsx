import React from 'react';
import Text from './Text';
import { mountWithTheme } from '../Tools/testHelper';

const text = 'Neque porro quisquam est qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit...';

describe('Text', () => {
  it('should render correctly', () => {
    const tree = mountWithTheme(<Text>{text}</Text>);

    expect(tree.find(Text)).toMatchSnapshot();
  });

  it('should render correctly (size)', () => {
    const tree = mountWithTheme(<Text size="size5">{text}</Text>);

    expect(tree.find(Text)).toMatchSnapshot();
  });

  it('should render correctly (weight)', () => {
    const tree = mountWithTheme(<Text weight="weight9">{text}</Text>);

    expect(tree.find(Text)).toMatchSnapshot();
  });
});
