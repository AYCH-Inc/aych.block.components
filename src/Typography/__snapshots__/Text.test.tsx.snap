// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should render correctly (size) 1`] = `
.c0 {
  font-family: Inter,sans-serif;
  font-size: 1rem;
  font-weight: 400;
  color: #353F52;
}

<Text
  size="size5"
  weight="weight4"
>
  <styled.span
    size="size5"
    weight="weight4"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  font-family: Inter, sans-serif;
  font-size: ",
              [Function],
              ";
  font-weight: ",
              [Function],
              ";
  color: ",
              [Function],
              ";
",
            ],
          },
          "displayName": "styled.span",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "span",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      size="size5"
      weight="weight4"
    >
      <span
        className="c0"
        size="size5"
      >
        Neque porro quisquam est qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit...
      </span>
    </StyledComponent>
  </styled.span>
</Text>
`;

exports[`Text should render correctly (weight) 1`] = `
.c0 {
  font-family: Inter,sans-serif;
  font-size: 1rem;
  font-weight: 900;
  color: #353F52;
}

<Text
  size="size5"
  weight="weight9"
>
  <styled.span
    size="size5"
    weight="weight9"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  font-family: Inter, sans-serif;
  font-size: ",
              [Function],
              ";
  font-weight: ",
              [Function],
              ";
  color: ",
              [Function],
              ";
",
            ],
          },
          "displayName": "styled.span",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "span",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      size="size5"
      weight="weight9"
    >
      <span
        className="c0"
        size="size5"
      >
        Neque porro quisquam est qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit...
      </span>
    </StyledComponent>
  </styled.span>
</Text>
`;

exports[`Text should render correctly 1`] = `
.c0 {
  font-family: Inter,sans-serif;
  font-size: 1rem;
  font-weight: 400;
  color: #353F52;
}

<Text
  size="size5"
  weight="weight4"
>
  <styled.span
    size="size5"
    weight="weight4"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  font-family: Inter, sans-serif;
  font-size: ",
              [Function],
              ";
  font-weight: ",
              [Function],
              ";
  color: ",
              [Function],
              ";
",
            ],
          },
          "displayName": "styled.span",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "span",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      size="size5"
      weight="weight4"
    >
      <span
        className="c0"
        size="size5"
      >
        Neque porro quisquam est qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit...
      </span>
    </StyledComponent>
  </styled.span>
</Text>
`;
